File: ios/Classes/ModelViewerPlusPlugin.swift
import Flutter
import UIKit

public class ModelViewerPlusPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let factory = ModelViewerPlusViewFactory(messenger: registrar.messenger())
        registrar.register(factory, withId: "model_viewer_plus")
    }
}

File: android/src/main/kotlin/com/example/model_viewer_plus/ModelViewerPlusPlugin.kt
package com.example.model_viewer_plus

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.BinaryMessenger

/** ModelViewerPlusPlugin */
class ModelViewerPlusPlugin: FlutterPlugin {
    private lateinit var factory: ModelViewerPlusFactory

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        val messenger: BinaryMessenger = flutterPluginBinding.binaryMessenger
        factory = ModelViewerPlusFactory(messenger)
        flutterPluginBinding.platformViewRegistry.registerViewFactory("model_viewer_plus", factory)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}
